Solutions/NonReducible.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * This class is responsible for reducing the string as much as possible
+ * 
+ * @author Kumar Kaushal
+ *
+ */
+public class NonReducible {
+	
+	/**
+	 * operation Count for reducing the string
+	 */
+	private int operationCount;
+	/**
+	 * final reduced string
+	 */
+	private String reducedString;
+		
+	public NonReducible() {
+		
+	}
+	
+	public NonReducible(int operationCount, String reducedString) {
+		this.operationCount = operationCount;
+		this.reducedString = reducedString;
+	}
+
+	public int getOperationCount() {
+		return operationCount;
+	}
+
+	public void setOperationCount(int operationCount) {
+		this.operationCount = operationCount;
+	}
+
+	public String getReducedString() {
+		return reducedString;
+	}
+
+	public void setReducedString(String reducedString) {
+		this.reducedString = reducedString;
+	}
+
+	/**
+	 * This method takes the input parameter, reduce it as much as possible and
+	 * also find out the count of operations needed to reduce it.
+	 * 
+	 * @param stringToBeReduced string to be reduced
+	 * @return NonReducible instance with reduced string and operation count set in that
+	 * 
+	 */
+	public NonReducible findReducedStringAndOperationCount(String stringToBeReduced) {
+		this.operationCount = 0;
+		String reducibleString = "";
+		NonReducible nonReducible = new NonReducible();
+
+		if (stringToBeReduced != null && !stringToBeReduced.isEmpty()) {
+			for (int i = 0; i < stringToBeReduced.length() - 1;) {
+				char[] inputCharArr = stringToBeReduced.toCharArray();
+
+				if (inputCharArr[i] == inputCharArr[i + 1]) {
```

Comment:
What will happen if I give a single char has input?

Reply:
Ideally same single char should be returned. Handled this case and modified the code accordingly.

Reply:
will it not crash?